src/rt/scheduler.rs

```diff
@@ -120,6 +118,17 @@ impl Scheduler {
             threads[thread.as_usize()].resume();
         });
     }
+
+    fn with_state<F, R>(f: F) -> R
+    where
+        F: FnOnce(&mut State<'_>) -> R,
+    {
+        if !STATE.is_set() {
+            panic!("cannot access Loom execution state from outside a Loom model. \
+            are you accessing a synchronization primitive from outside a Loom test (a call to `model` or `check`)?")
```

Comment:
nitpick, not important: perhaps we should add:
```suggestion
            are you accessing a Loom synchronization primitive from outside a Loom test (a call to `model` or `check`)?")
```

Reply:
Good idea; done!